utils/results/output/resultwriter.go

```diff
@@ -0,0 +1,317 @@
+	case format.Json:
+		if rw.printExtended {
+			return PrintJson(rw.commandResults)
+		}
+		return PrintJson(rw.commandResults.GetScaScansXrayResults())
+	case format.Sarif:
+		return rw.printSarif()
+	}
+	return nil
+}
+
+func (rw *ResultsWriter) createResultsConvertor(pretty bool) *conversion.CommandResultsConvertor {
+	return conversion.NewCommandResultsConvertor(conversion.ResultConvertParams{
+		IsMultipleRoots:              rw.isMultipleRoots,
+		IncludeLicenses:              rw.includeLicenses,
+		IncludeVulnerabilities:       rw.includeVulnerabilities,
+		Pretty:                       pretty,
+		AllowResultsWithoutLocations: true,
+	})
+}
+
+func (rw *ResultsWriter) printSarif() error {
+	sarifContent, err := rw.createResultsConvertor(false).ConvertToSarif(rw.commandResults)
+	if err != nil {
+		return err
+	}
+	sarifFile, err := sarifutils.ConvertSarifReportToString(sarifContent)
+	if err != nil {
+		return err
+	}
+	log.Output(sarifFile)
+	return nil
+}
+
+func PrintJson(output interface{}) (err error) {
+	results, err := utils.GetAsJsonString(output, true, true)
+	if err != nil {
+		return
+	}
+	log.Output(results)
+	return nil
+}
+
+func (rw *ResultsWriter) printTables() (err error) {
+	tableContent, err := rw.createResultsConvertor(isPrettyOutputSupported()).ConvertToTable(rw.commandResults)
+	if err != nil {
+		return
+	}
+	printMessages(rw.messages)
+	if rw.commandResults.HasInformation() {
+		var resultsPath string
+		if resultsPath, err = writeJsonResults(rw.commandResults); err != nil {
+			return
+		}
+		printMessage(coreutils.PrintTitle("The full scan results are available here: ") + coreutils.PrintLink(resultsPath))
+	}
+	log.Output()
+
+	if shouldPrintTable(rw.subScansPreformed, utils.ScaScan, rw.scanType) {
+		if rw.includeVulnerabilities {
+			err = PrintVulnerabilitiesTable(tableContent, rw.scanType, len(rw.commandResults.GetTechnologies()) > 0, rw.printExtended)
+		} else {
+			err = PrintViolationsTable(tableContent, rw.scanType, rw.printExtended)
+		}
+		if err != nil {
+			return
+		}
+		if rw.includeLicenses {
+			if err = PrintLicensesTable(tableContent, rw.printExtended, rw.scanType); err != nil {
+				return
+			}
+		}
+	}
+	if shouldPrintTable(rw.subScansPreformed, utils.SecretsScan, rw.scanType) {
+		if err = PrintJasTable(tableContent, rw.commandResults.EntitledForJas, jasutils.Secrets); err != nil {
+			return
+		}
+	}
+	if shouldPrintTable(rw.subScansPreformed, utils.IacScan, rw.scanType) {
+		if err = PrintJasTable(tableContent, rw.commandResults.EntitledForJas, jasutils.IaC); err != nil {
+			return
+		}
+	}
+	if !shouldPrintTable(rw.subScansPreformed, utils.SastScan, rw.scanType) {
+		return nil
+	}
+	return PrintJasTable(tableContent, rw.commandResults.EntitledForJas, jasutils.Sast)
+}
+
+// PrintVulnerabilitiesTable prints the vulnerabilities in a table.
+// Set printExtended to true to print fields with 'extended' tag.
+// If the scan argument is set to true, print the scan tables.
```

Comment:
There is no such an argument (not even in the original func).
It seems like it is an old logic or something that is in some inner struct. Please explain/ remove if not needed

Reply:
this is a real feature that we support. It allows you to print extended table or a default one if not specified.
![image](https://github.com/user-attachments/assets/58b1f381-b48a-4e87-a682-ccb628e9d27e)
 
This is an ecosystem Infrastructure that allows you to specify in the object how each attribute convert to values in a table to print just by adding tags, similar to `json`, as explained in the comment.
![image](https://github.com/user-attachments/assets/f1958648-28f3-421f-841f-cb6d49f13427)